src/Umbraco.Web.UI.Client/src/views/content/overlays/save.controller.js

```diff
@@ -58,57 +58,47 @@
             vm.variants = $scope.model.variants;
             vm.availableVariants = vm.variants.filter(saveableVariantFilter);
+            vm.isNew = vm.variants.some(variant => variant.state === 'NotCreated');
 
-            if(!$scope.model.title) {
-                localizationService.localize("content_readyToSave").then(function(value){
+            if (!$scope.model.title) {
+                localizationService.localize("content_readyToSave").then(value => {
                     $scope.model.title = value;
                 });
             }
 
-            _.each(vm.variants,
-                function (variant) {
+            vm.variants.forEach(variant => {
 
-                    //reset state:
-                    variant.save = false;
-                    variant.publish = false;
+                //reset state:
+                variant.save = variant.publish = false;
+                variant.isMandatory = isMandatoryFilter(variant);
 
-                    variant.isMandatory = isMandatoryFilter(variant);
-
-                    if(variant.state !== "NotCreated"){
-                        vm.isNew = false;
-                    }
-                });
-
-            _.each(vm.variants,
-                function (variant) {
-                    if(vm.isNew && hasAnyData(variant)){
-                        variant.save = true;
-                    }
-                });
+                if(vm.isNew && hasAnyData(variant)){
+                    variant.save = true;
+                }
+            });
 
             if (vm.variants.length !== 0) {
-
-                _.find(vm.variants, function (v) {
-                    if(v.active) {
-                        //ensure that the current one is selected
-                        v.save = true;
-                    }
-                });
-
-                vm.availableVariants.sort(function (a, b) {
+                        
+                //ensure that the current one is selected
+                var active = vm.variants.find(v => v.active);
+                if (active) {
+                    active.save = true;
+                }
+
+                vm.availableVariants.sort((a, b) => {
                     if (a.language && b.language) {
-                        if (a.language.name > b.language.name) {
+                        if (a.language.name < b.language.name) {
                             return -1;
                         }
-                        if (a.language.name < b.language.name) {
+                        if (a.language.name > b.language.name) {
                             return 1;
                         }
                     }
                     if (a.segment && b.segment) {
-                        if (a.segment > b.segment) {
+                        if (a.segment < b.segment) {
                             return -1;
                         }
-                        if (a.segment < b.segment) {
+                        if (a.segment > b.segment) {
```

Comment:
I'm seeing this reversal a few times @nathanwoulfe - that's scary! Can you explain? I don't get why it's the other way around now..

Reply:
Hey Seb, I'll double check to make sure I wasn't dreaming, but this change fixes the sort order of the variants listed in the save/publish overlay. They were previously in reverse alpha order, this flips them to ascending order.

Reply:
Would be nice actually to add this as a filter, given the same code is being used in a few places

Reply:
@nathanwoulfe There's a lot of repetition going on here for sure and it's super fragile that this code is copy/pasted all over the place!

Reply:
Might look into adding some filters. Other option might be to import common stuff as es6 modules to avoid the angular coupling. 

I'll see what I can make work.

Reply:
>  fixes the sort order of the variants listed in the save/publish overlay

Well whaddayaknow! That totally fixes it haha! 👍  


I'll merge this for now, refactoring can be done in a new PR! 🎉